--- a/net/minecraft/world/item/component/KineticWeapon.java
+++ b/net/minecraft/world/item/component/KineticWeapon.java
@@ -109,10 +_,7 @@
             double baseMobDamage = livingEntity.getAttributeBaseValue(Attributes.ATTACK_DAMAGE);
             boolean affected = false;
 
-            for (EntityHitResult hitResult : (Collection)ProjectileUtil.getHitEntitiesAlong(
-                    livingEntity, attackRange, e -> PiercingWeapon.canHitEntity(livingEntity, e), ClipContext.Block.COLLIDER
-                )
-                .map(a -> List.of(), e -> e)) {
+            for (EntityHitResult hitResult : ProjectileUtil.getHitEntitiesAlong(livingEntity, attackRange, e -> PiercingWeapon.canHitEntity(livingEntity, e), ClipContext.Block.COLLIDER).map(a -> List.<EntityHitResult>of(), e -> e)) {
                 Entity otherEntity = hitResult.getEntity();
                 if (otherEntity instanceof EnderDragonPart dragonPart) {
                     otherEntity = dragonPart.parentMob;
