--- a/net/minecraft/server/dedicated/DedicatedServerProperties.java
+++ b/net/minecraft/server/dedicated/DedicatedServerProperties.java
@@ -271,7 +_,7 @@
             if (worldPreset.is(WorldPresets.FLAT)) {
                 RegistryOps<JsonElement> ops = registries.createSerializationContext(JsonOps.INSTANCE);
                 Optional<FlatLevelGeneratorSettings> parsedSettings = FlatLevelGeneratorSettings.CODEC
-                    .parse(new Dynamic(ops, this.generatorSettings()))
+                    .parse(new Dynamic<>(ops, this.generatorSettings()))
                     .resultOrPartial(DedicatedServerProperties.LOGGER::error);
                 if (parsedSettings.isPresent()) {
                     return worldDimensions.replaceOverworldGenerator(registries, new FlatLevelSource(parsedSettings.get()));
