--- a/net/minecraft/core/HolderLookup.java
+++ b/net/minecraft/core/HolderLookup.java
@@ -41,7 +_,7 @@
         }
 
         default <V> RegistryOps<V> createSerializationContext(final DynamicOps<V> parent) {
-            return RegistryOps.create((DynamicOps<T>)parent, this);
+            return RegistryOps.create(parent, this);
         }
 
         static HolderLookup.Provider create(final Stream<HolderLookup.RegistryLookup<?>> lookups) {
