--- a/net/minecraft/core/Registry.java
+++ b/net/minecraft/core/Registry.java
@@ -38,7 +_,7 @@
         Codec<Holder.Reference<T>> referenceCodec = Identifier.CODEC
             .comapFlatMap(
                 name -> this.get(name)
-                    .<DataResult>map(DataResult::success)
+                    .map(DataResult::success)
                     .orElseGet(() -> DataResult.error(() -> "Unknown registry key in " + this.key() + ": " + name)),
                 holder -> holder.key().identifier()
             );
