--- a/net/minecraft/util/datafix/fixes/EntityRidingToPassengersFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityRidingToPassengersFix.java
@@ -65,7 +_,7 @@
                         Pair<String, Pair<Either<OldEntityTree, Unit>, Entity>> updating = input;
 
                         while (true) {
-                            Either<List<NewEntityTree>, Unit> passengersValue = (Either<List<NewEntityTree>, Unit>)DataFixUtils.orElse(
+                            Either<List<NewEntityTree>, Unit> passengersValue = DataFixUtils.orElse(
                                 passenger.map(
                                     p -> {
                                         Typed<NewEntityTree> newEntity = (Typed<NewEntityTree>)newEntityTreeType.pointTyped(ops)
@@ -78,15 +_,13 @@
                                 ),
                                 Either.right(DSL.unit())
                             );
-                            passenger = Optional.of(
-                                Pair.of(References.ENTITY_TREE.typeName(), Pair.of(passengersValue, ((Pair)updating.getSecond()).getSecond()))
-                            );
-                            Optional<OldEntityTree> riding = ((Either)((Pair)updating.getSecond()).getFirst()).left();
+                            passenger = Optional.of(Pair.of(References.ENTITY_TREE.typeName(), Pair.of(passengersValue, updating.getSecond().getSecond())));
+                            Optional<OldEntityTree> riding = updating.getSecond().getFirst().left();
                             if (riding.isEmpty()) {
                                 return passenger.orElseThrow(() -> new IllegalStateException("Should always have an entity tree here"));
                             }
 
-                            updating = (Pair<String, Pair<Either<OldEntityTree, Unit>, Entity>>)new Typed(oldEntityTreeType, ops, riding.get())
+                            updating = new Typed<>(oldEntityTreeType, ops, riding.get())
                                 .getOptional(entityTreeFinder)
                                 .orElseThrow(() -> new IllegalStateException("Should always have an entity here"));
                         }
