--- a/net/minecraft/util/datafix/fixes/EntityAttributeBaseValueFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityAttributeBaseValueFix.java
@@ -27,7 +_,7 @@
         return tag.update("attributes", attributes -> tag.createList(attributes.asStream().map(attribute -> {
             String attributeId = NamespacedSchema.ensureNamespaced(attribute.get("id").asString(""));
             if (!attributeId.equals(this.attributeId)) {
-                return (Dynamic)attribute;
+                return attribute;
             } else {
                 double base = attribute.get("base").asDouble(0.0);
                 return attribute.set("base", attribute.createDouble(this.valueFixer.applyAsDouble(base)));
