--- a/net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity.java
@@ -123,7 +_,7 @@
         this.litTimeRemaining = input.getShortOr("lit_time_remaining", (short)0);
         this.litTotalTime = input.getShortOr("lit_total_time", (short)0);
         this.recipesUsed.clear();
-        this.recipesUsed.putAll(input.<Map>read("RecipesUsed", RECIPES_USED_CODEC).orElse(Map.of()));
+        this.recipesUsed.putAll(input.read("RecipesUsed", RECIPES_USED_CODEC).orElse(Map.of()));
     }
 
     protected void saveAdditional(final ValueOutput output) {
