--- a/net/minecraft/world/entity/ai/sensing/VillagerHostilesSensor.java
+++ b/net/minecraft/world/entity/ai/sensing/VillagerHostilesSensor.java
@@ -8,7 +_,7 @@
 import net.minecraft.world.entity.ai.memory.MemoryModuleType;
 
 public class VillagerHostilesSensor extends NearestVisibleLivingEntitySensor {
-    private static final ImmutableMap<EntityType<?>, Float> ACCEPTABLE_DISTANCE_FROM_HOSTILES = ImmutableMap.builder()
+    private static final ImmutableMap<EntityType<?>, Float> ACCEPTABLE_DISTANCE_FROM_HOSTILES = ImmutableMap.<EntityType<?>, Float>builder()
         .put(EntityTypes.DROWNED, 8.0F)
         .put(EntityTypes.EVOKER, 12.0F)
         .put(EntityTypes.HUSK, 8.0F)
