--- a/net/minecraft/resources/RegistryOps.java
+++ b/net/minecraft/resources/RegistryOps.java
@@ -81,7 +_,7 @@
                     ? registryOps.lookupProvider
                         .lookup(registryKey)
                         .flatMap(r -> r.getter().get(key))
-                        .<DataResult<E>>map(DataResult::success)
+                        .map(DataResult::success)
                         .orElseGet(() -> DataResult.error(() -> "Can't find value: " + key))
                     : DataResult.error(() -> "Not a registry ops")
             )
