--- a/net/minecraft/util/CubicSpline.java
+++ b/net/minecraft/util/CubicSpline.java
@@ -61,7 +_,7 @@
             subSplineCodec -> Codec.either(Codec.FLOAT, CubicSpline.Multipoint.codec(coordinateCodec, subSplineCodec))
                 .xmap(e -> (CubicSpline)e.map(CubicSpline.Constant::new, m -> m), spline -> {
                     Objects.requireNonNull(spline);
-                    CubicSpline selector1$temp = spline;
+                    CubicSpline<I> selector1$temp = spline;
 
                     while (true) {
                         Either var10000;
