--- a/net/minecraft/util/ExtraCodecs.java
+++ b/net/minecraft/util/ExtraCodecs.java
@@ -291,14 +_,14 @@
             P max = (P)l.get(1);
             return makeInterval.apply(min, max);
         }), p -> ImmutableList.of(getMin.apply((I)p), getMax.apply((I)p)));
-        Codec<I> objectCodec = RecordCodecBuilder.create(
+        Codec<I> objectCodec = RecordCodecBuilder.<Pair<P, P>>create(
                 i -> i.group(pointCodec.fieldOf(lowerBoundName).forGetter(Pair::getFirst), pointCodec.fieldOf(upperBoundName).forGetter(Pair::getSecond))
                     .apply(i, Pair::of)
             )
             .comapFlatMap(p -> makeInterval.apply((P)p.getFirst(), (P)p.getSecond()), i -> Pair.of(getMin.apply((I)i), getMax.apply((I)i)));
         Codec<I> arrayOrObjectCodec = Codec.withAlternative(arrayCodec, objectCodec);
         return Codec.either(pointCodec, arrayOrObjectCodec)
-            .comapFlatMap(either -> (DataResult)either.map(min -> makeInterval.apply((P)min, (P)min), DataResult::success), p -> {
+            .comapFlatMap(either -> either.map(min -> makeInterval.apply((P)min, (P)min), DataResult::success), p -> {
                 P min = getMin.apply((I)p);
                 P max = getMax.apply((I)p);
                 return Objects.equals(min, max) ? Either.left(min) : Either.right(p);
@@ -605,7 +_,7 @@
             }
 
             private <T, V2 extends V> DataResult<T> encode(final Codec<V2> codec, final V input, final DynamicOps<T> ops) {
-                return codec.encodeStart(ops, input);
+                return codec.encodeStart(ops, (V2) input);
             }
         };
     }
