--- a/net/minecraft/world/entity/animal/cow/Cow.java
+++ b/net/minecraft/world/entity/animal/cow/Cow.java
@@ -53,7 +_,7 @@
     protected void readAdditionalSaveData(final ValueInput input) {
         super.readAdditionalSaveData(input);
         VariantUtils.readVariant(input, Registries.COW_VARIANT).ifPresent(this::setVariant);
-        input.<ResourceKey>read("sound_variant", ResourceKey.codec(Registries.COW_SOUND_VARIANT))
+        input.read("sound_variant", ResourceKey.codec(Registries.COW_SOUND_VARIANT))
             .flatMap(soundVariant -> this.registryAccess().lookupOrThrow(Registries.COW_SOUND_VARIANT).get((ResourceKey<CowSoundVariant>)soundVariant))
             .ifPresent(this::setSoundVariant);
     }
