--- a/net/minecraft/world/level/gameevent/EntityPositionSource.java
+++ b/net/minecraft/world/level/gameevent/EntityPositionSource.java
@@ -52,13 +_,13 @@
     }
 
     private void resolveEntity(final Level level) {
-        ((Optional)this.entityOrUuidOrId
+        this.entityOrUuidOrId
                 .map(
                     Optional::of,
                     uuidOrId -> Optional.ofNullable(
                         (Entity)uuidOrId.map(uuid -> level instanceof ServerLevel serverLevel ? serverLevel.getEntity(uuid) : null, level::getEntity)
                     )
-                ))
+                )
             .ifPresent(entity -> this.entityOrUuidOrId = Either.left(entity));
     }
 
