From eac7ff1dfe7e6095aeedfd8a5af6cf601689a028 Mon Sep 17 00:00:00 2001
From: Oliver <oschi1303@gmail.com>
Date: Thu, 2 Apr 2026 21:40:03 +0200
Subject: [PATCH 2/2] Origami entities

---
 net/minecraft/server/level/ServerPlayer.java | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/net/minecraft/server/level/ServerPlayer.java b/net/minecraft/server/level/ServerPlayer.java
index 9b46622..4f25084 100644
--- a/net/minecraft/server/level/ServerPlayer.java
+++ b/net/minecraft/server/level/ServerPlayer.java
@@ -228,6 +228,7 @@ public class ServerPlayer extends Player {
     );
     private static final boolean DEFAULT_SEEN_CREDITS = false;
     private static final boolean DEFAULT_SPAWN_EXTRA_PARTICLES_ON_FALL = false;
+    private final com.origamimc.origami.player.OrigamiPlayerImpl origamiPlayer; // Origami - origami player
     public ServerGamePacketListenerImpl connection;
     private final MinecraftServer server;
     public final ServerPlayerGameMode gameMode;
@@ -369,6 +370,7 @@ public class ServerPlayer extends Player {
 
     public ServerPlayer(final MinecraftServer server, final ServerLevel level, final GameProfile gameProfile, final ClientInformation clientInformation) {
         super(level, gameProfile);
+        this.origamiPlayer = com.origamimc.origami.player.OrigamiPlayerImpl.of(this); // Origami - origami player
         this.server = server;
         this.textFilter = server.createTextFilterForPlayer(this);
         this.gameMode = server.createGameModeForPlayer(this);
@@ -380,6 +382,12 @@ public class ServerPlayer extends Player {
         this.object = null;
     }
 
+    // Origami start - origami player
+    public com.origamimc.origami.player.OrigamiPlayerImpl getOrigamiPlayer() {
+        return origamiPlayer;
+    }
+    // Origami end
+
     public BlockPos adjustSpawnLocation(final ServerLevel level, final BlockPos spawnSuggestion) {
         CompletableFuture<Vec3> future = PlayerSpawnFinder.findSpawn(level, spawnSuggestion);
         this.server.managedBlock(future::isDone);
