--- a/net/minecraft/world/level/chunk/PalettedContainer.java
+++ b/net/minecraft/world/level/chunk/PalettedContainer.java
@@ -49,7 +_,7 @@
     private static <T, C extends PalettedContainerRO<T>> Codec<C> codec(
         final Codec<T> elementCodec, final Strategy<T> strategy, final T defaultValue, final PalettedContainerRO.Unpacker<T, C> unpacker
     ) {
-        return RecordCodecBuilder.create(
+        return RecordCodecBuilder.<PalettedContainerRO.PackedData>create(
                 i -> i.group(
                         elementCodec.mapResult(ExtraCodecs.orElsePartial(defaultValue))
                             .listOf()
