--- a/net/minecraft/world/entity/animal/camel/CamelAi.java
+++ b/net/minecraft/world/entity/animal/camel/CamelAi.java
@@ -50,7 +_,7 @@
             Activity.CORE,
             0,
             ImmutableList.of(
-                new Swim(0.8F),
+                new Swim<>(0.8F),
                 new CamelAi.CamelPanic(4.0F),
                 new LookAtTargetSink(45, 90),
                 new MoveToTargetSink(),
@@ -68,7 +_,7 @@
                 Pair.of(1, new AnimalMakeLove(EntityType.CAMEL)),
                 Pair.of(
                     2,
-                    new RunOne(
+                    new RunOne<>(
                         ImmutableList.of(
                             Pair.of(new FollowTemptation(camel -> 2.5F, camel -> camel.isBaby() ? 2.5 : 3.5), 1),
                             Pair.of(BehaviorBuilder.triggerIf(Predicate.not(Camel::refuseToMove), BabyFollowAdult.create(ADULT_FOLLOW_RANGE, 2.5F)), 1)
@@ -78,7 +_,7 @@
                 Pair.of(3, new RandomLookAround(UniformInt.of(150, 250), 30.0F, 0.0F, 0.0F)),
                 Pair.of(
                     4,
-                    new RunOne(
+                    new RunOne<>(
                         ImmutableMap.of(MemoryModuleType.WALK_TARGET, MemoryStatus.VALUE_ABSENT),
                         ImmutableList.of(
                             Pair.of(BehaviorBuilder.triggerIf(Predicate.not(Camel::refuseToMove), RandomStroll.stroll(2.0F)), 1),
