--- a/net/minecraft/util/datafix/schemas/V704.java
+++ b/net/minecraft/util/datafix/schemas/V704.java
@@ -141,7 +_,7 @@
     });
     protected static final HookFunction ADD_NAMES = new HookFunction() {
         public <T> T apply(final DynamicOps<T> ops, final T value) {
-            return V99.addNames(new Dynamic(ops, value), V704.ITEM_TO_BLOCKENTITY, V99.ITEM_TO_ENTITY);
+            return V99.addNames(new Dynamic<>(ops, value), V704.ITEM_TO_BLOCKENTITY, V99.ITEM_TO_ENTITY);
         }
     };
 
