--- a/net/minecraft/world/entity/monster/warden/AngerManagement.java
+++ b/net/minecraft/world/entity/monster/warden/AngerManagement.java
@@ -67,10 +_,8 @@
 
     private List<Pair<UUID, Integer>> createUuidAngerPairs() {
         return Streams.concat(
-                new Stream[]{
-                    this.suspects.stream().map(e -> Pair.of(e.getUUID(), this.angerBySuspect.getInt(e))),
-                    this.angerByUuid.object2IntEntrySet().stream().map(e -> Pair.of((UUID)e.getKey(), e.getIntValue()))
-                }
+                this.suspects.stream().map(e -> Pair.of(e.getUUID(), this.angerBySuspect.getInt(e))),
+                this.angerByUuid.object2IntEntrySet().stream().map(e -> Pair.of((UUID)e.getKey(), e.getIntValue()))
             )
             .collect(Collectors.toList());
     }
